### PR TITLE
Draft documentation guidelines

### PR DESCRIPTION
**Pull request by chosak**
Adds new guide for developers about writing documentation.

Ping @niqjohnson who I can't request as a reviewer here.

### CODE REVIEW

File: guides/documentation.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Documenting your work
+
+Good development practices include writing and maintaining documentation that covers the software being written.
+
+The Write the Docs [Documentation Guide](https://www.writethedocs.org/guide/) provides an excellent writeup on [why writing documentation is important](https://www.writethedocs.org/guide/writing/beginners-guide-to-docs/#why-write-docs), best processes for writing and maintaining docs, and tips for building a documentation culture in an organization.
+
+The WTD [Documentation Principles](https://www.writethedocs.org/guide/writing/docs-principles/) are also recommended reading and should be kept in mind when writing documentation at CFPB.
```
Comment (Scotchester) [inline guides/documentation.md]: What would you think about putting this before the links to their guide homepage and the why-it's-important writeup? If I want people to read one thing before tl;dr-ing out, it's the principles.

File: guides/documentation.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Documentation definition of done
+
+Any new or modified documentation should meet the following criteria before being published:
+
+1. _Thoroughly reviewed:_ Ideally, documentation should be peer-reviewed by two people: one who is familiar with the material being covered, and one who is new to the material. This will ensure that documentation is appropriate for developers with different experience levels.
+1. _Tested:_ Any documentation that describes a process (for example, setting up a project) should be tested fully by a developer to verify its correctness. Consider defining and using user stories to guide this process (for example, "as a new member of the team, I need to get cf.gov running locally on my machine").
+1. _Properly located:_ Documentation should be located in an appropriate place, and be discoverable for developers seeking it out. See [above](#where-to-put-documentation) for additional detail.
```
[the remark this answers is not in the document]
Comment (niqjohnson) [reply guides/documentation.md]: Should we also add a bit here about linking to any new documentation from the right places, like from the development repo's `README.md` if you add something new there or from the documentation landing page if you add something that's important enough to go there?

File: guides/documentation.md (hunk 1 above)
Comment (chosak) [reply guides/documentation.md]: Thanks, that's a good suggestion. Implemented in 429a507.

File: guides/documentation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chosak) [reply guides/documentation.md]: Great suggestions, both. Applied in 429a507.

**Comment by ascott1** [comment]
:heart: this

File: guides/documentation.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+Any new or modified documentation should meet the following criteria before being published:
+
+1. _Thoroughly reviewed:_ Ideally, documentation should be peer-reviewed by two people: one who is familiar with the material being covered, and one who is new to the material. This will ensure that documentation is appropriate for developers with different experience levels.
+1. _Tested:_ Any documentation that describes a process (for example, setting up a project) should be tested fully by a developer to verify its correctness. Consider defining and using user stories to guide this process (for example, "as a new member of the team, I need to get cf.gov running locally on my machine").
+1. _Properly located:_ Documentation should be located in an appropriate place. [See "Where to put documentation" above](#where-to-put-documentation) for additional detail.
+1. _Properly discoverable:_ Documentation should be linked to from an appropriate landing page (for example, this repository's [`README.md`](https://github.com/cfpb/development/blob/master/README.md) and, if possible, be discoverable through search ([for example](https://cfpb.github.io/cfgov-refresh/search.html?q=testing)).
```
Comment (Scotchester) [inline guides/documentation.md]:
```suggestion
1. _Properly discoverable:_ Documentation should be linked to from an appropriate landing page (for example, [this repository's README](https://github.com/cfpb/development/blob/master/README.md) and, if possible, be discoverable through search (like [the cfgov-refresh docs search](https://cfpb.github.io/cfgov-refresh/search.html?q=testing), for example).
```

Comment (chosak) [reply guides/documentation.md]: Applied (with semantic line breaks -- thanks for teaching me [about that](http://sembr.org/)) in 6080ed4.

**Pull request merged**